cmd/upgrade.go

```diff
@@ -176,53 +183,72 @@ func (uc *upgradeCmd) loadCluster(cmd *cobra.Command) error {
 	if !found {
 		return errors.Errorf("upgrading from Kubernetes version %s to version %s is not supported. To see a list of available upgrades, use 'aks-engine get-versions --version %s'", uc.containerService.Properties.OrchestratorProfile.OrchestratorVersion, uc.upgradeVersion, uc.containerService.Properties.OrchestratorProfile.OrchestratorVersion)
 	}
+	return nil
+}
 
-	// Read the name suffix from the parameters to identify VMs in the resource group that belong to this cluster.
-	templatePath := path.Join(uc.deploymentDirectory, "azuredeploy.json")
-	contents, err := ioutil.ReadFile(templatePath)
-	if err != nil {
-		return errors.Wrap(err, "error reading ARM file")
+func (uc *upgradeCmd) validateCurrentLocalState(armTemplateHandle io.Reader) error {
+	if uc.containerService.Location == "" {
```

Comment:
Let's put `uc.containerService.Location` to a variable.

Reply:
Yeah, I just didn't touch that code as there is no tests around it. I just extracted it to a func.
I'll leave it as is for this PR if that's ok.